packages/client/src/runtime/RequestHandler.ts

```diff
@@ -46,14 +46,16 @@ export type Request = {
 function getRequestInfo(requests: Request[]) {
   const txId = requests[0].transactionId
   const inTx = requests[0].runInTransaction
-  const headers = requests[0].headers
+  const headers = requests[0].headers ?? {}
 
   // if the tx has a number for an id, then it's a regular batch tx
   const _inTx = typeof txId === 'number' && inTx ? true : undefined
   // if the tx has a string for id, it's an interactive transaction
   const _txId = typeof txId === 'string' && inTx ? txId : undefined
 
-  return { inTx: _inTx, headers: { transactionId: _txId, ...headers } }
+  if (_txId !== undefined) headers.transactionId = _txId
```

Comment:
Fixes an issue with miniflare, remix, and others where `fetch` is injected with undici's implementation for local development. The problem was that `transactionId` was being passed `undefined`, which caused runtime errors.

Reply:
https://github.com/prisma/prisma/issues/12356 I guess

Reply:
Correct, I should have done this in the isolated bugs PR.